### PR TITLE
Add math_acos_or_asin function

### PR DESCRIPTION
**Pull request by S-H-GAMELINKS**
`math_acos` and `math_asin` function has similar code in `math.c`.
Add `match_acos_or_asin` function to refacter these function.

### CODE REVIEW

File: math.c [hunk 1]
```diff
@@ -179,14 +179,20 @@ math_tan(VALUE unused_obj, VALUE x)
...
 
 static VALUE
-math_acos(VALUE unused_obj, VALUE x)
+math_acos_or_asin(VALUE x, double func(double))
 {
     double d;
 
     d = Get_Double(x);
     /* check for domain error */
     if (d < -1.0 || 1.0 < d) domain_error("acos");
```
Comment (Maumagnaguagno) [inline math.c:188]: The argument of ``domain_error`` is different for acos and asin.

Comment (S-H-GAMELINKS) [reply math.c:188]: Thanks! I'll fixed it.

**Comment by nobu** [comment]
It doesn't look simpler to me.

**Comment by S-H-GAMELINKS** [comment]
On second thoughts this pr was not simple code.
Close it.

**Pull request closed without merging**